client/lib/wp/sync-handler/cache.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+import debugFactory from 'debug';
+import ms from 'ms';
+import negate from 'lodash/negate';
+import matchesProperty from 'lodash/matchesProperty';
+import filter from 'lodash/filter';
+
+/**
+ * Internal dependencies
+ */
+import { getLocalForage } from 'lib/localforage';
+
+/**
+ * Module variables
+ */
+const localforage = getLocalForage();
+const debug = debugFactory( 'calypso:sync-handler:cache' );
+
+const debugPromise = ( ...args ) => {
+	return new Promise( resolve => {
+		debug.call( this, ...args );
+		resolve();
+	} );
+};
+
+const RECORDS_LIST_KEY = 'records-list';
+const SYNC_RECORD_REGEX = /^sync-record-\w+$/;
+const LIFETIME = '2 days';
+
+/**
+ * Check it the given key is a `sync-record-` key
+ *
+ * @param {String} key - record key
+ * @return {Boolean} `true` if it's a sync-record-<key>
+ */
+const isSyncRecordKey = key => SYNC_RECORD_REGEX.test( key );
+
+export class Cache {
+	getAll() {
+		return localforage.getItem( RECORDS_LIST_KEY );
+	}
+
+	/**
+	 * Add the given `key` into the records-list object
+	 * adding at the same a timestamp (now).
+	 * If the pair key-timestamp already exists it will be updated.
+	 *
+	 * @param {String} key - record key
+	 * @return {Promise} promise
+	 */
+	addItem( key ) {
+		return this.dropByKey( key ).then( records => {
+			debug( 'adding %o to list (%s)', key, records.length + 1 );
+			return localforage.setItem( RECORDS_LIST_KEY, [
+				...records,
+				{ key, timestamp: Date.now() }
+			] );
+		} );
+	}
+
+	removeItem( key ) {
+		return this.dropByKey( key ).then( records => {
+			return localforage.setItem( RECORDS_LIST_KEY, records );
+		} );
+	}
+
+	/**
+	 * Retrieve all records filter by the given key
+	 *
+	 * @param {String} key - compare records with this key
+	 * @return {Promise} promise
+	 */
+	dropByKey( key ) {
```

Comment:
yeah ... I thought that because I just added another method named `dropOlderThan` and maybe it could be useful for us to expose them like Class methods  in a near future. But we should't concern about the future, right ? :sunglasses: